pkg/apis/gcp/validation/infrastructure.go

```diff
@@ -124,6 +124,10 @@ func ValidateInfrastructureConfig(infra *apisgcp.InfrastructureConfig, nodesCIDR
 		}
 	}
 
+	if infra.Networks.CloudNAT != nil && len(infra.Networks.CloudNAT.NatIPNames) == 0 {
```

Comment:
Correct me if I am wrong but this validation will also start failing for valid InfrastructureConfig such as:
```yaml
networks:
  cloudNAT:
    minPortsPerVM: 2048
```

We should rather require at least 1 nat IP names to be provided **only** when the field is explicitly set to empty array.
We should not require at least 1 nat IP name to be provided **always** because the field is optional.

The check we need is:
```suggestion
	if infra.Networks.CloudNAT != nil && infra.Networks.CloudNAT.NatIPNames != nil && len(infra.Networks.CloudNAT.NatIPNames) == 0 {
```

because `len(nil)` is `0`.

Can you also add an unit test to cover this scenario?